stac_fastapi/api/stac_fastapi/api/openapi.py

```diff
@@ -1,68 +1,36 @@
 
 
 def update_openapi(app: FastAPI) -> FastAPI:
     """Update OpenAPI response content-type.
 
     This function modifies the openapi route to comply with the STAC API spec's
-    required content-type response header
+    required content-type response header.
     """
-    urls = (server_data.get("url") for server_data in app.servers)
-    server_urls = {url for url in urls if url}
-
-    async def openapi(req: Request) -> JSONResponse:
-        root_path = req.scope.get("root_path", "").rstrip("/")
-        if root_path not in server_urls:
-            if root_path and app.root_path_in_servers:
-                app.servers.insert(0, {"url": root_path})
-                server_urls.add(root_path)
-        return VndOaiResponse(app.openapi())
-
-    # Remove the default openapi route
-    app.router.routes = list(
-        filter(lambda r: r.path != app.openapi_url, app.router.routes)
+    # Find the route for the openapi_url in the app.
+    openapi_route: Route = next(
+        route for route in app.router.routes if route.path == app.openapi_url
     )
-    # Add the updated openapi route
-    app.add_route(app.openapi_url, openapi, include_in_schema=False)
-    return app
 
-
-# TODO: Remove or fix, this is currently unused
-# and calls a missing method on ApiSettings
-def config_openapi(app: FastAPI, settings: ApiSettings):
-    """Config openapi."""
-
-    def custom_openapi():
-        """Config openapi."""
-        if app.openapi_schema:
-            return app.openapi_schema
-
-        openapi_schema = get_openapi(
-            title="Arturo STAC API", version="0.1", routes=app.routes
-        )
-
-        if settings.api_extension_is_enabled(ApiExtensions.fields):
-            openapi_schema["paths"]["/search"]["get"]["responses"]["200"]["content"][
-                "application/json"
-            ]["schema"] = {"$ref": "#/components/schemas/ItemCollection"}
-            openapi_schema["paths"]["/search"]["post"]["responses"]["200"]["content"][
-                "application/json"
-            ]["schema"] = {"$ref": "#/components/schemas/ItemCollection"}
-
-        app.openapi_schema = openapi_schema
-        return app.openapi_schema
-
-    app.openapi = custom_openapi
+    # Create a patched endpoint that modifies the content type of the response
+    async def patched_openapi_endpoint(req: Request) -> Response:
+        # Get the response from the original endpoint
+        response: JSONResponse = await openapi_route.endpoint(req)
+        # Update the content type header in place
+        response.headers[
+            "content-type"
+        ] = "application/vnd.oai.openapi+json;version=3.0"
+        # Return the updated response
+        return response
+
+    # When a route is accessed the `handle` function will call `self.app`. So we can
+    # leave the `endpoint` the original function for use in the patched function and
+    # just update the `app` to use the patched function.
```

Comment:
Is this a documented use-case? If so, can you drop a link in a reply just so I can read up and make myself smart? Thanks!

Reply:
Unfortunately this is not something I found any documentation on. I had to work my way through fastapi / starlette to figure out the process. I was attempting to document my thought process so someone else can follow what is happening in case modifications are needed down the line. I may have summarized too succinctly. 

My first attempt was to just modify the `openapi_route.endpoint` which seemed liked the obvious place to start. Unfortunately, that was unsuccessful and required much more digging. You can even set `endpoint = None` and nothing changes. 

What I figured out:
- The `openapi_route` is a `starlette.Route`
- When it is instantiated it does some checks to see if `endpoint` is a function / method and then wraps it with `request_response` and sets that as the `app`
- When the route is called it will call the `handle` function.
- When `handle` is called it only uses `self.app` so the `endpoint` function has no bearing on the output. 

So I just called the existing `endpoint` from the patched function. Wrapped it with `request_response` and set that as the `app`. Thinking about it more, it might be slightly better to do `old_endpoint = openapi_route.endpoint` and call that. Then actually update the endpoint on the route to be the patched function. The `__eq__` function does use it to compare routes.